auth/src/main/java/com/firebase/ui/auth/ui/provider/GitHubLoginActivity.java

```diff
@@ -0,0 +1,92 @@
+package com.firebase.ui.auth.ui.provider;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.RestrictTo;
+import android.support.customtabs.CustomTabsIntent;
+import android.support.v4.content.ContextCompat;
+
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.remote.GitHubSignInHandler;
+import com.firebase.ui.auth.ui.HelperActivityBase;
+import com.firebase.ui.auth.util.ExtraConstants;
+
+@SuppressLint("GoogleAppIndexingApiWarning")
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class GitHubLoginActivity extends HelperActivityBase {
+    private static final String REFRESH_ACTION = "refresh_action";
+
+    private boolean mShouldCloseCustomTab;
+
+    @NonNull
+    public static Intent createIntent(Context context, Uri starter) {
+        return new Intent(context, GitHubLoginActivity.class)
+                .putExtra(ExtraConstants.PARAMS, starter);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (savedInstanceState == null) {
+            new CustomTabsIntent.Builder()
+                    .setShowTitle(true)
+                    .enableUrlBarHiding()
+                    .setToolbarColor(ContextCompat.getColor(this, R.color.colorPrimary))
+                    .build()
+                    .launchUrl(this, (Uri) getIntent().getParcelableExtra(ExtraConstants.PARAMS));
```

Comment:
We're casting the whole `PARAMS` to a `Uri`?  Maybe it should come with a different `ExtraConstants` key?